tldw_Server_API/app/core/MCP_unified/protocol.py

```
@@ -698,6 +717,166 @@ def _hash_arguments(arguments: dict[str, Any]) -> Optional[str]:
+    ) -> Optional[str]:
+        """Normalize idempotency key from explicit argument or request params."""
+        raw_idempotency_key = idempotency_key
+        if raw_idempotency_key is None:
+            raw_idempotency_key = params.get("idempotencyKey")
+            if raw_idempotency_key is None:
+                raw_idempotency_key = params.get("idempotency_key")
+
+        if raw_idempotency_key is None:
+            return None
+        if not isinstance(raw_idempotency_key, str):
+            raise InvalidParamsException("idempotencyKey must be a string")
```

Ruff (0.15.7)

[warning] Avoid specifying long messages outside the exception class

(TRY003)